src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int amountInStore = FruitStorage.fruitRepository.get(fruit);
+        int amountToBuy = fruitTransaction.getQuantity();
+        if (amountInStore < amountToBuy) {
+            throw new RuntimeException("Not enough fruit in the storage for this operation");

```

Comment:
The exception message should include more context, such as the type of fruit and the requested amount vs. the amount available in storage. This will make debugging and logging easier.